mmengine/optim/optimizer/optimizer_wrapper.py

```diff
@@ -163,15 +161,17 @@ def __init__(self,
         # last few iterations. If `_max_counts` has not been initialized,
         # the loss factor will always be the same as `_accumulative_counts`.
         self._remainder_counts = -1
+
         self.new_param_settings = {
-            'params': torch.tensor([0.0], requires_grad=True),
+            'params': torch.tensor([0.0],
+                                   requires_grad=True,
+                                   dtype=torch.float),
             'is_state_tracker': True
         }
 
-
-        if hasattr(self.optimizer, 'defaults'):
-            self.new_param_settings.update(**self.optimizer.defaults)
-            self.optimizer.param_groups.append(self.new_param_settings)
+        # if hasattr(self.optimizer, 'defaults'):
+        self.new_param_settings.update(**self.optimizer.defaults)
+        self.optimizer.param_groups.append(self.new_param_settings)
```

Comment:
PyTorch's optimizer performs some efficiency optimizations with parameter group updates. I believe that directly adding a parameter group to the optimizer could potentially lead to conflicts. Instead, we can modify the OptimWrapper to include an additional param_groups that tracks the learning rate parameter group, without affecting the optimizer's original parameter groups.

Reply:
Thanks for your suggestions. We appreciate your efforts in modifying the logic to append the `tracker` at the optimizer wrappers level in order to avoid conflicts. We hope that the current implementation is more efficient and user-friendly. If you have any further questions or need additional assistance, please let us know.